libraries/rush-lib/src/logic/operations/BuildTimePlugin.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+import { OperationStatus } from './OperationStatus';
+
+const PLUGIN_NAME: 'BuildTimePlugin' = 'BuildTimePlugin';
+
+interface ITimelineRecord {
+  startTime: number;
+  duration: number;
+  name: string;
+  status: OperationStatus;
+}
+
+/**
+ * @beta
+ */
+export interface IBuildTimeRecord {
+  project: string;
+  buildTime: number;
+  status: OperationStatus;
+}
+
+export class BuildTimePlugin implements IPhasedCommandPlugin {
+  private _buildTimes: IBuildTimeRecord[];
+
+  public constructor() {
+    this._buildTimes = [];
+  }
+
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        this._buildTimes = _setBuildTimes(result);
+      }
+    );
+  }
+}
+
+/**
+ * @beta
+ */
+export function _setBuildTimes(result: IExecutionResult): IBuildTimeRecord[] {
+  const data: ITimelineRecord[] = [];
+  const buildTimes: IBuildTimeRecord[] = [];
+
+  for (const [operation, operationResult] of result.operationResults) {
+    if (operation.runner?.silent) {
+      continue;
+    }
+
+    const { stopwatch } = operationResult;
+
+    const { startTime, endTime } = stopwatch;
+
+    if (startTime && endTime) {
+      const { duration } = stopwatch;
+
+      data.push({
+        startTime,
+        duration,
+        name: operation.associatedProject?.packageName ?? '',
```

Comment:
Using the package name instead of the operation name artificially limits the utility of this feature. Test it out on running `rush retest` within the rushstack repository and you will see the problem immediately.